File: src/CmdGroup.cpp
// CmdGroup.C

#include "CmdGroup.H"
#include <QDebug>

static CBuilder<CmdGroup> cbGroup("group");

bool CmdGroup::usage() {
  return error("Usage: group\n");
}

bool CmdGroup::parse(Statement const &s) {
  if (s.length()==1)
    return true;
  else
    return usage();
}

void CmdGroup::render(Statement const &, Figure &f, bool) {
  f.startGroup();
}

File: backend/python/cmdgroup.py
# cmdgroup.py - This file is part of QPlot

# QPlot - Publication quality 2D graphs with dual coordinate systems
# Copyright (C) 2014  Daniel Wagenaar
# (same GPL licence as original)

from __future__ import annotations

from typing import TYPE_CHECKING

from command import Command

if TYPE_CHECKING:
    from statement import Statement
    from figure import Figure


@Command.register("group")
class CmdGroup(Command):

    def _usage(self) -> bool:
        return self.error("Usage: group\n")

    def parse(self, s: Statement) -> bool:
        return True if len(s) == 1 else self._usage()

    def render(self, s: Statement, f: Figure, dryrun: bool) -> None:
        f.start_group()
